suzieq/poller/worker/services/service.py

```diff
@@ -748,6 +758,32 @@ async def run(self):
             if output:
                 ostatus = [x.get('status', -1) for x in output]
 
+                hostname = output[0]["hostname"]
+                namespace = output[0]["namespace"]
+                key = f'{namespace}-{hostname}'
+
+                should_commit = True
+
+                # we should always commit during the first run of the service
+                # for each node regardless their status.
+                if not self.is_first_run(key):
+                    # If it's not the first run and some command did fail,
+                    # increment the count to be used in hysteresis.
+                    # The info will be committed in a future run.
+                    if any((not Service.is_status_ok(x) for x in ostatus)):
+                        self._consecutive_failures[key] = \
+                            self._consecutive_failures.get(key, 0) + 1
+                    # otherwise reset the counter.
+                    else:
+                        self._consecutive_failures[key] = 0
+
+                if 0 < self._consecutive_failures.get(key, 0) < \
+                        HYSTERESIS_FAILURE_CNT:
+                    self.logger.info(
+                        f"{self.name} {key} node failure hysteresis,"
+                        "skipping data commit")
+                    should_commit = False
+
```

Comment:
Are we checking if this is the case due to specific failures such as timeout and ssh connection failed as opposed to any failure? We should get into hysterisis only on connectivity failures.

Reply:
It's not easy to detect connectivity failures. From the tests I've done, status code can be -1, 2 and very rarely 408.
However -1 is the generic status code we use for an unknown failure so we might end up using hysteresis with non connectivity issues.